resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/PerformanceCriticalCodeAnalysisStage.cs

```diff
@@ -359,7 +412,10 @@ public override void VisitEqualityExpression(IEqualityExpression equalityExpress
 
             public bool InteriorShouldBeProcessed(ITreeNode element, HotMethodAnalyzerContext context)
             {
-                return !(element is ICSharpClosure);
+                if (element is ICSharpClosure closure)
```

Comment:
It looks like this code should process local functions when their invocations are found.
Is it intended to ignore costly calls within lambda / anonymous method / query expressions? 
e.g.
```cs
void M()
{
  Action a = () => CostlyInvocation();
  a();
}
```

Reply:
I haven't seen any code that will mark local function invocations (or any closures, altough it seems that any othe forms of closures aren't processed) as visited as the `MarkCurrentAsVisited` method is only called within `VisitMethodDeclaration`. Therefore they'll probably be analyzed multiple times. While it won't cause stackoverflow exception due to the depth limit it can cause a huge performance hit especially in case of large recursive functions since they will be analyzed `MaxDepth` times for each call. Please check performance for code like this:

```cs
void M()
{
  Trigger(); // start inspecting the local function

  void Trigger()
  {
    Local(); // inspect the second local function
    Trigger(); // inspect Trigger and consequently Local again MaxDepth times
  }

  void Local()
  {
    // 1000+ lines of code here, inspect them all
    Local(); // inspect Local MaxDepth times recursively
  }
}
```

Will this analyze the `Local` source `MaxDepth`^2 times?